src/state/routing/slice.ts

```diff
@@ -0,0 +1,111 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { isPlainObject } from '@reduxjs/toolkit'
+import { createApi, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react'
+import { Protocol } from '@uniswap/router-sdk'
+// Importing just the type, so smart-order-router is lazy-loaded
+// eslint-disable-next-line no-restricted-imports
+import type { ChainId } from '@uniswap/smart-order-router'
+import ms from 'ms.macro'
+import qs from 'qs'
+
+import { GetQuoteResult } from './types'
+
+const protocols: Protocol[] = [Protocol.V2, Protocol.V3]
+
+// routing API quote query params: https://github.com/Uniswap/routing-api/blob/main/lib/handlers/quote/schema/quote-schema.ts
+const DEFAULT_QUERY_PARAMS = {
+  protocols: protocols.map((p) => p.toLowerCase()).join(','),
+}
+
+const serializeRoutingCacheKey = ({ endpointName, queryArgs }: { endpointName: string; queryArgs: any }) => {
```

Comment:
Is this still necessary, with serializableCheck turned off?

Reply:
yes - we still need to write a new serializable fxn in order to generate a query cache key; we can now just safely silence the errors about storing the serializable provider in state

Reply:
could you instead just call` defaultSerializeQueryArgs` from redux toolkit, but pass in your args w/o provider?

```
defaultSerializeQueryArgs( ({ provider, ...rest}) => ({...rest})())
```